server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java

```diff
@@ -638,9 +688,19 @@ public void testNegativeZero() {
             NumberType.HALF_FLOAT.rangeQuery("field", null, +0f, true, false, false, MOCK_CONTEXT, isIndexed)
         );
 
-        assertNotEquals(NumberType.DOUBLE.termQuery("field", -0d, isIndexed), NumberType.DOUBLE.termQuery("field", +0d, isIndexed));
-        assertNotEquals(NumberType.FLOAT.termQuery("field", -0f, isIndexed), NumberType.FLOAT.termQuery("field", +0f, isIndexed));
-        assertNotEquals(NumberType.HALF_FLOAT.termQuery("field", -0f, isIndexed), NumberType.HALF_FLOAT.termQuery("field", +0f, isIndexed));
+        final boolean hasDocValues = isIndexed == false || randomBoolean(); // at least one should be true
+        assertNotEquals(
+            NumberType.DOUBLE.termQuery("field", -0d, isIndexed, hasDocValues),
+            NumberType.DOUBLE.termQuery("field", +0d, isIndexed, hasDocValues)
+        );
+        assertNotEquals(
+            NumberType.FLOAT.termQuery("field", -0f, isIndexed, hasDocValues),
+            NumberType.FLOAT.termQuery("field", +0f, isIndexed, hasDocValues)
+        );
+        assertNotEquals(
+            NumberType.HALF_FLOAT.termQuery("field", -0f, isIndexed, hasDocValues),
+            NumberType.HALF_FLOAT.termQuery("field", +0f, isIndexed, hasDocValues)
+        );
```

Comment:
general question about the test changes: do we need a test to verify the new behaviour? I was not clear if you added it already, or if these were just adjutments required by the prod changes. Wouldn't we ideally check that we produce an IndexOrDocValuesQuery in the places where we introduced that behaviour?

Reply:
I have improved the test testTermQuery to cover all cases in [935b7d7](https://github.com/elastic/elasticsearch/pull/128293/commits/935b7d789ec51f653ab0102aaeb1a4ce156e5dea)